Utils/PoolableObject.cs

```diff
@@ -0,0 +1,24 @@
+﻿using DUCK.Utils;
+using System;
+using UnityEngine;
+
+namespace DUCK.Pooling
+{
+	public class PoolableObject : MonoBehaviour
+	{
+		public event Action OnAddedToPool;
+		public event Action OnRemovedFromPool;
+
+		public void AddToPool()
+		{
+			OnAddedToPool.SafeInvoke();
+		}
+
+		public void RemoveFromPool()
+		{
+			OnRemovedFromPool.SafeInvoke();
+		}
+
+		public void CopyFrom(PoolableObject instance) { }
```

Comment:
Why is this needed? To avoid specifically being able to copy a camera?

If so I'd expect CopyFrom to still work but to interact with the pool.

Reply:
No - to mimic the behaviour of Unity's Instantiate<T>(T original) - see ObjectPool.cs:67

Reply:
In that case it might be worth making this abstract or throw. Right now if I'm meant to expect `pool.Instantiate(someObject); ` to work out of the box I may not know that I need to implement my own CopyFrom.